p2p/metrics.go

```diff
@@ -67,23 +67,22 @@ func markDialError(err error) {
 	if !metrics.Enabled() {
 		return
 	}
-	if err2 := errors.Unwrap(err); err2 != nil {
```

Comment:
I think what we really want here is

```go
var disc DiscReason
if errors.As(err, &disc) {
    switch disc {
    case ...
    }
}
```

Reply:
Sounds reasonable, and also cleaner. I will verify how it works, and come back with a patch.

Reply:
The `dialProtoHandshakeError` would have to be in an else branch, and we would still need to handle `dialEncHandshakeError` separately.
I would say the original with `Is` and `As` is better. I've added a catch-all and some comments to make it even cleaner.